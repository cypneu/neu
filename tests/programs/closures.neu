fn make_adder(x) {
    fn adder(y) {
        return x + y;
    }
    return adder;
}
add5 = make_adder(5);
add10 = make_adder(10);
print(add5(3));
print(add10(3));

fn make_counter() {
  i = 0;
  fn count() {
    i = i + 1;
    return i;
  }
  return count;
}
c1 = make_counter();
c2 = make_counter();
print(c1());
print(c1());
print(c2());

fn make_adder(n) {
    fn add(x) { return n + x; }
    return add;
}

add5 = make_adder(5);
res  = add5(3);
print(res);

fn outer() {
    x = 10;
    fn inner() { return x; }
    x = 20;
    return inner;
}

f   = outer();
res = f();
print(res);
