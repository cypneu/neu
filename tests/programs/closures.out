8
13
1
2
1
8
20
